util/codec/codec.go

```diff
@@ -833,6 +846,40 @@ func DecodeOne(b []byte) (remain []byte, d types.Datum, err error) {
 	return b, d, nil
 }
 
+// DecodeAsDateTime decodes on datum from []byte of `KindMysqlTime`.
+func DecodeAsDateTime(b []byte, tp byte, loc *time.Location) (remain []byte, d types.Datum, err error) {
+	if len(b) < 1 {
+		return nil, d, errors.New("invalid encoded key")
+	}
+	flag := b[0]
+	b = b[1:]
+	switch flag {
+	case uintFlag:
+		var v uint64
+		b, v, err = DecodeUint(b)
+		if err != nil {
+			return b, d, err
+		}
+		t := types.NewTime(types.ZeroCoreTime, mysql.TypeDatetime, 0)
+		err = t.FromPackedUint(v)
+		if err == nil {
+			if tp == mysql.TypeTimestamp && !t.IsZero() {
```

Comment:
Should we check if the loc is `nil` here? `&& loc != nil`.

Reply:
I agree with it.